gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/error/AbstractErrorHandler.java

```diff
@@ -66,9 +69,19 @@ public final void handle(RoutingContext routingContext) {
                 }
 
                 // append error information
-                parameters.set(ConstantKeys.ERROR_PARAM_KEY, "server_error");
+                parameters.set(ConstantKeys.ERROR_PARAM_KEY, SERVER_ERROR);
                 parameters.set(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY, "Unexpected error occurred");
 
+                String toHash = SERVER_ERROR + "$" + "Unexpected error occurred";
+                String hash = HashUtil.generateSHA256(toHash);
+                if (routingContext.session() != null) {
+                    if (!routingContext.session().isDestroyed()) {
+                        routingContext.session().put(ERROR_HASH, hash);
+                    } else {
+                        routingContext.session().put(ERROR_HASH, hash);
+                    }
```

Comment:
question: Are you sure about this one ? either the condition is not useful and you can add the hash everytime or a destoyed session can't receive a new value without an initialization.

Reply:
when the session is destroyed and there is .put() method called, the session will be recreated.